drivers/can/stm32_ll_can.c

```diff
@@ -0,0 +1,752 @@
+			mode_reg  |= bank_bit;
+			scale_reg |= bank_bit;
+			break;
+		case CAN_FILTER_EXTENDED_MASKED:
+			shift_width = old_filter_type == CAN_FILTER_STANDARD ? -3 : -1;
+			mode_reg  &= ~bank_bit;
+			scale_reg |= bank_bit;
+			break;
+		}
+		filter_index_tmp = can_calc_filter_index(filter_nr, mode_reg, scale_reg);
+		res = can_shift_arr(device_data->rx_response, filter_index_tmp + 1,
+					shift_width);
+		if (filter_index_tmp >= CAN_MAX_NUMBER_OF_FILTES || res) {
+			SYS_LOG_INF("No space for a new filter!");
+			filter_nr = CAN_NO_FREE_FILTER;
+			goto done;
+		}
+		can_shift_bits(&device_data->response_type, filter_index_tmp + 1,
+			       shift_width);
+		can->FM1R = mode_reg;
+		can->FS1R = scale_reg;
+	} else {
+		filter_index_tmp = can_calc_filter_index(filter_nr, can->FM1R,
+							 can->FS1R);
+		if (filter_index_tmp >= CAN_MAX_NUMBER_OF_FILTES) {
+			filter_nr = CAN_NO_FREE_FILTER;
+			goto done;
+		}
+	}
+	can_stm32_set_filter_bank(filter_nr, &can->sFilterRegister[bank_nr],
+				  filter_type, id, mask);
+done:
+	can->FA1R |= bank_bit;
+	can->FMR &= ~(CAN_FMR_FINIT);
+	SYS_LOG_DBG("Filter set! Filter number: %d (index %d)",
+		    filter_nr, filter_index_tmp);
+	*filter_index = filter_index_tmp;
+	return filter_nr;
+}
+
+
+static inline int can_stm32_attach(struct device *dev, void *response_ptr,
+				   struct can_filter *filter, int *filter_index)
+{
+	const struct can_stm32_config *cfg = DEV_CFG(dev);
+	struct can_stm32_data *data = DEV_DATA(dev);
+	CAN_TypeDef *can = cfg->can;
+	int filter_index_tmp = 0;
+	int filter_nr;
+
+	k_mutex_lock(&data->set_filter_mutex, K_FOREVER);
+	filter_nr = can_stm32_set_filter(filter, data, can, &filter_index_tmp);
+	if (filter_nr != CAN_NO_FREE_FILTER) {
+		data->rx_response[filter_index_tmp] = response_ptr;
+	}
+	k_mutex_unlock(&data->set_filter_mutex);
+	*filter_index = filter_index_tmp;
+	return filter_nr;
+}
+
+int can_stm32_attach_msgq(struct device *dev, struct k_msgq *msgq,
+			  struct can_filter *filter)
+{
+	int filter_nr;
+	int filter_index;
+	struct can_stm32_data *data = DEV_DATA(dev);
+
+	filter_nr = can_stm32_attach(dev, msgq, filter, &filter_index);
+	data->response_type |= (1ULL << filter_index);
+	return filter_nr;
+}
+
+int can_stm32_attach_isr(struct device *dev, can_rx_callback_t isr,
+			 struct can_filter *filter)
+{
+	int filter_nr;
+	int filter_index;
+	struct can_stm32_data *data = DEV_DATA(dev);
+
+	filter_nr = can_stm32_attach(dev, isr, filter, &filter_index);
+	data->response_type &= ~(1ULL << filter_index);
+	return filter_nr;
+}
+
+void can_stm32_detach(struct device *dev, int filter_nr)
+{
+	__ASSERT_NO_MSG(filter_nr >= 0 && filter_nr < CAN_MAX_NUMBER_OF_FILTES);
+	const struct can_stm32_config *cfg = DEV_CFG(dev);
+	struct can_stm32_data *data = DEV_DATA(dev);
+	CAN_TypeDef *can = cfg->can;
+	int bank_nr;
+	int filter_index;
+	u32_t bank_bit;
+	u32_t mode_reg;
+	u32_t scale_reg;
+	can_filter_type_t type;
+
+	k_mutex_lock(&data->set_filter_mutex, K_FOREVER);
```

Comment:
Next 20 lines are quite tightly packed, could you add some empty lines there so it would look a bit nicer to reader.